databricks_cli/__init__.py

```diff
@@ -24,6 +24,22 @@
 # pylint:disable=import-error
 # pylint:disable=bare-except
 
+import warnings
+
+
+def issue_deprecation_warning():
+    # Don't print deprecation warning when running the CLI itself.
+    import sys
+    import os
+    if sys.argv and os.path.basename(sys.argv[0]) == 'databricks':
+        return
+    warnings.warn("the databricks_cli module is deprecated in favor of databricks-sdk-py. Python "
+                  "3.12 will be the last version of Python supported by databricks-cli. Please "
+                  "migrate to databricks-sdk-py as documented in the migration guide: "
+                  "https://docs.databricks.com/en/dev-tools/cli/migrate.html",
```

Comment:
This migration guide doesn't mention the SDK, only the CLI.

We should link to SDK docs here, assuming callers are trying to leverage this code as SDK.

Reply:
Hm, we never made a migration guide for the Python SDK...